src/safeds/data/tabular/transformation/_discretizer.py

```diff
@@ -0,0 +1,205 @@
+from __future__ import annotations
+
+from sklearn.preprocessing import KBinsDiscretizer as sk_KBinsDiscretizer
+
+from safeds.data.tabular.containers import Table
+from safeds.data.tabular.transformation._table_transformer import TableTransformer
+from safeds.exceptions import NonNumericColumnError, TransformerNotFittedError, UnknownColumnNameError
+
+
+class Discretizer(TableTransformer):
+    """
+    The Discretizer bins continuous data into intervals.
+
+    Parameters
+    ----------
+    number_of_bins: float
+        The number of bins to be created.
+
+    Raises
+    ------
+    ValueError
+        If the given number_of_bins is less than 2.
+    """
+
+    def __init__(self, number_of_bins: float = 5):
+        self._column_names: list[str] | None = None
+        self._wrapped_transformer: sk_KBinsDiscretizer | None = None
+
+        if number_of_bins < 2:
+            raise ValueError("Parameter 'number_of_bins' must be >= 2.")
+        self._number_of_bins = number_of_bins
+
+    def fit(self, table: Table, column_names: list[str] | None) -> Discretizer:
+        """
+        Learn a transformation for a set of columns in a table.
+
+        This transformer is not modified.
+
+        Parameters
+        ----------
+        table : Table
+            The table used to fit the transformer.
+        column_names : Optional[list[str]]
```

Comment:
```suggestion
        column_names : list[str] | None
```
See https://github.com/Safe-DS/Stdlib/issues/374#issuecomment-1602555092